apps/src/feedback.js

```diff
@@ -502,6 +529,23 @@ FeedbackUtils.prototype.displayFeedback = function (options, requiredBlocks,
   }
 };
 
+/**
+ * Converts the image data uri to a blob, then saves it to the server as the
+ * thumbnail for the current project.
+ * @param {string} image Image encoded as a data URI.
+ * @returns {Promise} A promise which resolves if successful.
+ */
+FeedbackUtils.saveThumbnail = function (image) {
+  return dataURIToBlob(image)
+    .then(project.saveThumbnail)
+    .then(() => {
+      return new Promise(resolve => {
+        // Save the thumbnail url to the server.
+        project.save(resolve);
```

Comment:
OK, I don't know how you talked me into it but I've made `project.save()` return a Promise. This very special snippet of argument-shifting code has now been relegated to the dust heaps of history:
```
  save(...args) {
    let [sourceAndHtml, callback, forceNewVersion, preparingRemix] = args;
    ...
    if (typeof args[0] === 'function' || !sourceAndHtml) {
      // If no save data is provided, shift the arguments and ask for the
      // latest save data ourselves.
      [callback, forceNewVersion, preparingRemix] = args;
      ...
```